shared/src/writeToTypegenFile.ts

```diff
@@ -0,0 +1,32 @@
+import * as fs from "fs";
+import * as prettier from "prettier";
+import { promisify } from "util";
+import { getTypegenOutput } from ".";
+
+export const writeToTypegenFile = async (opts: {
+  filePath: string;
+  event: Parameters<typeof getTypegenOutput>[0];
+}) => {
+  const prettierConfig = await prettier.resolveConfig(opts.filePath);
+  const pathToSave = opts.filePath.replace(
+    /\.(mjs|js|tsx|ts|jsx)$/,
+    ".typegen.ts",
+  );
+
+  try {
+    if (opts.event.machines.some((machine) => machine.hasTypesNode)) {
+      const typegenOutput = getTypegenOutput(opts.event);
+      await promisify(fs.writeFile)(
+        pathToSave,
+        prettier.format(typegenOutput, {
+          ...prettierConfig,
+          parser: "typescript",
+        }),
+      );
+    } else if (await promisify(fs.exists)(pathToSave)) {
```

Comment:
Using `fs.exists` is deprecated and not recommended. The "blessed" way of doing most of the file-operations like this is to just do the thing and swallow the `ENOENT` error if it happens, while usually rethrowing other error types